api/v1alpha1/packagebundlecontroller_webhook_test.go

```diff
@@ -0,0 +1,156 @@
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	clientfake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func TestHandleInner(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("validates successfully", func(t *testing.T) {
+		v := &activeBundleValidator{}
+		pbc := &PackageBundleController{
+			Spec: PackageBundleControllerSpec{
+				ActiveBundle: "v1-21-1001",
+			},
+		}
+		bundles := &PackageBundleList{
+			Items: []PackageBundle{
+				{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "v1-21-1001",
+					},
+					Spec:   PackageBundleSpec{},
+					Status: PackageBundleStatus{},
+				},
+			},
+		}
+		resp, err := v.handleInner(ctx, pbc, bundles)
+		if assert.NoError(t, err) {
+			assert.NotNil(t, resp)
+			assert.True(t, resp.AdmissionResponse.Allowed)
+		}
+	})
+
+	t.Run("handles decoder errors", func(t *testing.T) {
+		v := &activeBundleValidator{}
+		req := admission.Request{
+			AdmissionRequest: admissionv1.AdmissionRequest{
+				Object: runtime.RawExtension{
+					Raw: nil,
+				},
+			},
+		}
+		resp := v.Handle(ctx, req)
+		assert.False(t, resp.Allowed)
+		assert.Contains(t, resp.Result.Message, "decoding request")
+	})
+
+	t.Run("handles list errors", func(t *testing.T) {
+		pbc := &PackageBundleController{
+			Spec: PackageBundleControllerSpec{
+				ActiveBundle: "v1-21-1001",
+			},
+			Status: PackageBundleControllerStatus{},
+		}
+		pbcBytes, err := json.Marshal(pbc)
+		require.NoError(t, err)
+		log.Printf("marshaled %d bytes", len(pbcBytes))
+		scheme := runtime.NewScheme()
+		require.NoError(t, AddToScheme(scheme))
+		decoder, err := admission.NewDecoder(scheme)
+		require.NoError(t, err)
+		v := &activeBundleValidator{
+			decoder: decoder,
+			Client:  newFakeClient(scheme),
+		}
+		req := admission.Request{
+			AdmissionRequest: admissionv1.AdmissionRequest{
+				Object: runtime.RawExtension{
+					Raw: pbcBytes,
+				},
+			},
+		}
+		resp := v.Handle(ctx, req)
+		assert.False(t, resp.Allowed)
+		assert.Contains(t, resp.Result.Message, "listing package bundles: testing error")
+	})
+
+	t.Run("rejects unknown bundle names", func(t *testing.T) {
+		v := &activeBundleValidator{}
+		pbc := &PackageBundleController{
+			Spec: PackageBundleControllerSpec{
+				ActiveBundle: "v1-21-1002",
+			},
+		}
+		bundles := &PackageBundleList{
+			Items: []PackageBundle{
+				{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "v1-21-1001",
+					},
+					Spec:   PackageBundleSpec{},
+					Status: PackageBundleStatus{},
+				},
+			},
+		}
+		resp, err := v.handleInner(ctx, pbc, bundles)
+		if assert.NoError(t, err) {
+			assert.False(t, resp.AdmissionResponse.Allowed)
+			assert.Equal(t, metav1.StatusFailure, resp.AdmissionResponse.Result.Status)
```

Comment:
```suggestion
			assert.Equal(t, metav1.StatusReasonNotFound, resp.AdmissionResponse.Result.Status)
```

Reply:
This isn't quite what we'd want.

The Status field and the Reason field are different things.

I can add your suggest, but my original code is still valid.